### PR TITLE
EDGECLOUD-2001 Add object detection using GPU-accelerated library

### PR DESCRIPTION
**Pull request by brucearmstrong**
This is the client side update to support an Object Detection activity.
Also:
- EDGECLOUD-2261 Demo app - Add backend info to "About" dialog
- EDGECLOUD-157 Demo app "About" dialog/page with links

### CODE REVIEW

File: android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java [hunk 1]
```diff
@@ -56,6 +56,7 @@
 public class ImageSender {
     private static final String TAG = "ImageSender";
     public static final int TRAINING_COUNT_TARGET = 10;
+    private static final double RECOGNITION_CONFIDENCE_THRESHOLD = 105;
```
Comment (franklin-huang-mobiledgex) [inline android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java]: Why 105?

Comment (franklin-huang-mobiledgex) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java]: Is it not on a 0-1 scale?

File: android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                if (mMirrored) {
+                    Log.d(TAG, "Mirroring!");
+                    // The image that was processed is what the camera sees, but the image we want to
+                    // overlay the rectangle onto is mirrored. So not only do we have to scale it,
+                    // but we have to flip it horizontally.
+                    rect.left = mWidth - rect.left;
+                    rect.right = mWidth - rect.right;
+                    int tmp = rect.left;
+                    rect.left = rect.right;
+                    rect.right = tmp;
```
Comment (franklin-huang-mobiledgex) [inline android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java]: I think we can simplify the mirroring (unless you think they other way is more intuitive):
int tmp = rect.left
rect.left = mWidth - rect.right;
rect.right = mWidth - tmp;

File: android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java (hunk 1 above)
Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java]: This has always been confusing. The "confidence" value is actually a "distance" from the trained image that the submitted image most closely matches. So "0" means a perfect match, but there is actually no upper limit. 

105 was arrived at through trial and error. Note that this threshold was previously on the server side, and if reached, the subject name returned was "Unknown". I decided to always return the matched subject name even if the confidence value isn't good, and let the client decide whether to show the subject name, "Unknown", or as we do now add "[Doubtful]" to the returned subject.

The highest value I've seen is over 200. Again, there's no actual upper limit, so it's difficult to come up with a "x% confidence" value.

File: android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    private int mHeightOff;
+    private float mServerToDisplayRatioX;
+    private float mServerToDisplayRatioY;
+    private boolean mMirrored;
+    private Animation mAlphaAnim;
+
+    /**
+     * Array for drawing each object with a different color.
+     */
+    private String[] colors = {
```
Comment (lgarner) [inline android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java]: Bit hard coded?

Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java]: I'm open to suggestions for an alternative means of having an array of pleasant colors.

Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java]: These are from https://matplotlib.org/examples/color/colormaps_reference.html, Qualitative, "tab10" set, followed by the 10 lighter variants from "tab20".

Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ObjectClassRenderer.java]: Your comment prompted me to add a check so we don't try to pull more colors than the array holds.

File: android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java (hunk 1 above)
Comment (lgarner) [reply android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java]: I guess the threshold is for some kind of curve where only the bottom part matters?

**Pull request merged**